not-a-toy.R

```diff
@@ -126,26 +144,33 @@ ap_prediction <- function(time, m0, n0, P, k) {
 # m: observed total methane values, same length as time
 # n: observed labeled methane values, same length as time
 # Returns the sum of squares between predicted and observed m and AP
-cost_function <- function(params, time, m, n) {
+cost_function <- function(params, time, m, n, sdD, sdC) {
     #    message(params["P"], ",", params["k"])
     pred <- ap_prediction(time = time,
                           m0 = m[1],
                           n0 = n[1],
                           P = params["P"],
-                          k = params["k"])
-
-    # m and n are on different scales, so we need to scale them
-    # in order to combine for a single sum of squares calculation
-    # First find overall ranges...
-    m_range <- range(c(pred$mt, m, na.rm = TRUE))
-    n_range <- range(c(pred$nt, n, na.rm = TRUE))
-    # ...and then rescale
-    mt_r <- rescale(pred$mt, from = m_range)
-    nt_r <- rescale(pred$nt, from = n_range)
-    m_r <- rescale(m, from = m_range)
-    n_r <- rescale(n, from = n_range)
-    # Return overall sum of squares to the optimizer
-    sum((c(mt_r, nt_r) - c(m_r, n_r)) ^ 2)
+                          k = params["k"]
+                          )
+
+    if(OLD_METHOD) {
+      # m and n are on different scales, so we need to scale them
+      # in order to combine for a single sum of squares calculation
+      # First find overall ranges...
+      m_range <- range(c(pred$mt, m, na.rm = TRUE))
+      n_range <- range(c(pred$nt, n, na.rm = TRUE))
+      # ...and then rescale
+      mt_r <- rescale(pred$mt, from = m_range)
+      nt_r <- rescale(pred$nt, from = n_range)
+      m_r <- rescale(m, from = m_range)
+      n_r <- rescale(n, from = n_range)
+      # Return overall sum of squares to the optimizer
+      sum((c(mt_r, nt_r) - c(m_r, n_r)) ^ 2)
+      
+    } else {
+      sum((abs(m - pred$mt)/sd(m))*Nm + (abs(n - pred$nt)/sd(n))*Nd)
```

Comment:
A comment would be useful here, explaining what's happening and referencing relevant vFH equation

Reply:
Done.